server/routes/projects.test.js

```diff
@@ -0,0 +1,515 @@
+const request = require('supertest');
+const app = require('./projects.js');
+
+jest.mock('../db/db-connection.js', () => ({
+    query: jest.fn(),
+}));
+
+const db = require('../db/db-connection.js');
+
+describe('GET /:user_id/projects', () => {
+    // resets mock function before each test
+    beforeEach(() => {
+        db.query.mockReset();
+    })
+
+    it('returns a list of projects by user', async () => {
+        // mocking the database query response
+        const projectList = [
+            {
+                created_at: '2024-01-01T00:00:00.000Z',
+                id: 101,
+                is_favorite: false,
+                project_name: 'Cat Loaf',
+                project_status: 'In Progress',
+                project_type: 'Amigurimi',
+                sentiment_score: null,
+                user_id: 20,
+                user_name: 'oliver'
+            },
+            {
+                created_at: '2024-02-01T00:00:00.000Z',
+                id: 102,
+                is_favorite: true,
+                project_name: 'Star Blanket',
+                project_status: 'Completed',
+                project_type: 'Blanket',
+                sentiment_score: null,
+                user_id: 20,
+                user_name: 'oliver'
+            }
+        ];
```

Comment:
Ah, I think this is one of the ones I missed since I made the test before switching to using the fixture as a mock database. I will note to update it when I add more tests for the nice-to-have features, since I do already have a mock database with this data.